include/sparrow/struct_array.hpp

```diff
@@ -242,7 +242,56 @@ namespace sparrow
          */
         [[nodiscard]] SPARROW_API array_wrapper* raw_child(std::size_t i);
 
-    private:
+        /**
+         * @brief Gets the names of all child arrays.
+         *
+         * @return Range of child array names.
+         */
+        [[nodiscard]] auto names() const
+        {
+            return get_arrow_proxy().children()
+                   | std::views::transform(
+                       [](const auto& child)
+                       {
+                           return child.name();
+                       }
+                   );
+        }
+
+        /**
+         * @brief Adds a child array to the struct.
+         *
+         * @param child The child array to add
+         *
+         * @post Increases the number of children by one
+         */
+        template <layout_or_array A>
+        void add_child(A&& child);
```

Comment:
Should this method accept a range instead (or provide an overload accepting a range) for performance considerations? (i.e. building the children list once for all instead of ubilding it at each add in a loop)